### PR TITLE
new tool: msms_extractor to extract MS/MS scans from mzML files based…

### PR DESCRIPTION
**Pull request by pravs3683**
New Tool: "**msms_extractor**" reads scan numbers from the PSM report (scan numbers with identified PSM) and gives the option to create a new subset mzml file. Uses Proteowizard's msconvert.
    
    Input: mzML file and PSM report file
    Output: mzML file with scan numbers that were not assigned any PSM.

### CODE REVIEW

File: tools/msms_extractor/msms_extractor.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# 
+# Developed by Praveen Kumar
+# Galaxy-P Team (Griffin's Lab)
+# University of Minnesota
+#
+#
+#
+
+def main():
+    from pyteomics import mzml
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.py]: import should go ontop of the file

File: tools/msms_extractor/msms_extractor.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    import sys
+    import shutil
+    import subprocess
+    import re
+    import pandas as pd
+    from operator import itemgetter
+    from itertools import groupby
+    import random
+    
+    if len(sys.argv) >= 7:
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.py]: please use argparse instead, that should be way more readable...

File: tools/msms_extractor/msms_extractor.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.xml]: empty line

File: tools/msms_extractor/msms_extractor.xml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+<tool id="msms_extractor" name="msms_extractor" version="1.0.0">
+  <description>Extract MS/MS scans from the mzML file(s) based on PSM report.</description>
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.xml]: remove dot

File: tools/msms_extractor/msms_extractor.xml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  <requirements>
+      <requirement type="package" version="3_0_9992">proteowizard</requirement>
+      <requirement type="package" version="3.4">pyteomics</requirement>
+  </requirements>
+  <command detect_errors="exit_code"><![CDATA[python msms_extractor.py $spectrumfile $psmreportfile $output ${spectrumfile.name.rsplit('.',1)[0]} $removeretain.doremoveretain $removeretain.num_random_scans]]></command>
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.xml]: please use `$__tool_directory__/msms_extractor.py` and single-quote all text and data-parameters.

File: tools/msms_extractor/msms_extractor.xml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  <inputs>
+    <param name="spectrumfile" type="data" format="mzml">
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.xml]: labels can go directly into the param tag

File: tools/msms_extractor/msms_extractor.xml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        <param name="doremoveretain" type="select" display="radio" label="Retain or Remove the MS/MS scans that were identified in PSM report">
+            <option value="remove" selected="True">Remove the scans reported in the PSM Report File</option>
+            <option value="retain">Retain the scans reported in the PSM Report File </option>
+        </param>
+        
+        <when value="remove">
+            <param name="num_random_scans" type="hidden" value="0" />
+        </when>
+        <when value="retain">
+            <param name="num_random_scans" type="integer" label="Add N random scans in addition to those in the list" value="0" optional="false" />
```
Comment (bgruening) [inline tools/msms_extractor/msms_extractor.xml]: if you only have one param in a conditional you can make this param optional="true"

Comment (pravs3683) [reply tools/msms_extractor/msms_extractor.xml]: Thanks @bgruening ! I have made all the changes as suggested. Thanks a lot again!

**Pull request merged**